lib/Twig/Extension/Core.php

```diff
@@ -852,6 +852,39 @@ function twig_in_filter($value, $compare)
     return false;
 }
 
+/** 
+ * Returns a trimmed string.
+ * 
+ * @param string $string
+ * @param string $characterMask
+ * @param string $mode
+ *
+ * @return string
+ *
+ * @throws Twig_Error_Runtime When an type is not a string.
+ * @throws Twig_Error_Runtime When an invalid type is used (not 'left', 'right' or 'both')
+ */
+function twig_trim_filter($string, $characterMask = null, $type = 'both')
+{
+    if (!is_string($type)) {
+        throw new Twig_Error_Runtime(sprintf('Trimming type must be a string but got %s.', gettype($type)));
+    }
+    if (!in_array($type, ['both', 'left', 'right'])) {
+        throw new Twig_Error_Runtime(sprintf('Trimming type must be "left", "right" or "both" but got %s.', $type));
+    }
+    if (null === $characterMask) {
+        $characterMask = " \t\n\r\0\x0B";
+    }
```

Comment:
One last CS thingy, please add a blank line after a control structure.
Beside these few minors this looks good to me, thanks for adding this, much appreciated  :+1:  :)